### PR TITLE
Feat unknownclass + some fix

### PR DESCRIPTION
**Pull request by xiphl**
Added data primitive: UNKNOWN.
1. UI edit pages need to allow for new data type
2. ingest-api needs to allow for new data type

Removed dependency on adhocConfig.ts for ingest-api. There will still be 2 other URLs that will be hardcoded during CI/CD, but those are deemed ok.

Added a minor lint fix for tableau.py

### CODE REVIEW

**Comment by xiphl** [comment]
damn my brain is really not awake, why did i only look for port 3000 and not port 9002

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/BrowsePath/EditBrowsePathTable.tsx [hunk 1]
```diff
@@ -28,10 +28,24 @@ function timeout(delay: number) {
...
+    // for UI edit pages, the URL is complicated, need to find the root path.
+    const mainPathLength = initialUrl.split('/', 3).join('/').length;
+    const mainPath = `${initialUrl.substring(0, mainPathLength + 1)}`;
+    let publishUrl = initialUrl.includes(':3000')
+        ? initialUrl.replace(':3000/', ':8001/custom/update_browsepath')
+        : mainPath;
+    publishUrl = publishUrl.includes(':9002')
+        ? publishUrl.replace(':9002/', ':8001/custom/update_browsepath')
+        : `${mainPath}/custom/update_browsepath`;
+    console.log(`the final url is ${publishUrl}`);
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/BrowsePath/EditBrowsePathTable.tsx:44]: I guess we should refactor these code to a function

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Delete/DeleteSchemaTabv2.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { GetDatasetQuery } from '../../../../../../graphql/dataset.generated';
 // import { useGetAuthenticatedUser } from '../../../../../useGetAuthenticatedUser';
 import { useBaseEntity } from '../../../EntityContext';
 import { FindMyUrn, FindWhoAmI, GetMyToken } from '../../../../dataset/whoAmI';
-import adhocConfig from '../../../../../../conf/Adhoc';
+// import adhocConfig from '../../../../../../conf/Adhoc';
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Delete/DeleteSchemaTabv2.tsx:10]: remove unused import statement

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Delete/DeleteSchemaTabv2.tsx [hunk 3]
```diff
@@ -27,9 +27,26 @@ function CheckStatus(entity) {
...
+    publishUrl = publishUrl.includes(':9002')
+        ? publishUrl.replace(':9002/', ':8001/custom/update_dataset_status')
+        : `${mainPath}/custom/update_dataset_status`;
+    // const publishUrl = mainPath.includes(':3000')
+    //     ? mainPath.replace(':3000/', ':8001/custom/update_dataset_status')
+    //     : `${mainPath}/custom/update_dataset_status`;
+    console.log(`the final url is ${publishUrl}`);
+    // let url = adhocConfig;
+    // const branch = url.lastIndexOf('/');
+    // url = `${url.substring(0, branch)}/update_dataset_status`;
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Delete/DeleteSchemaTabv2.tsx:49]: reuse the function

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/EditPropertiesTableEditable.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import React, { useState } from 'react';
 import { Button, Divider, Form, Input, message, Table, Typography } from 'antd';
 import PropTypes from 'prop-types';
 import axios from 'axios';
-import adhocConfig from '../../../../../../conf/Adhoc';
+// import adhocConfig from '../../../../../../conf/Adhoc';
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/EditPropertiesTableEditable.tsx:6]: remove unused imports and comments pls

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/EditPropertiesTableEditable.tsx [hunk 5]
```diff
@@ -13,9 +13,23 @@ import { FindMyUrn, FindWhoAmI, GetMyToken } from '../../../../dataset/whoAmI';
...
+    let publishUrl = mainPath.includes(':3000')
+        ? mainPath.replace(':3000/', ':8001/custom/update_properties')
+        : mainPath;
+    publishUrl = mainPath.includes(':9002')
+        ? mainPath.replace(':9002/', ':8001/custom/update_properties')
+        : `${mainPath}/custom/update_properties`;
+    console.log(`the final url is ${publishUrl}`);
+    // let url = adhocConfig;
+    // const branch = url.lastIndexOf('/');
+    // url = `${url.substring(0, branch)}/update_properties`;
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/EditPropertiesTableEditable.tsx:32]: reuse the code pls

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/samples/EditSampleForm.tsx [hunk 6]
```diff
@@ -35,10 +35,20 @@ function timeout(delay: number) {
...
+    // having a setup that works for both makes development easier.
+    // for UI edit pages, the URL is complicated, need to find the root path.
+    const mainPathLength = initialUrl.split('/', 3).join('/').length;
+    const mainPath = `${initialUrl.substring(0, mainPathLength + 1)}`;
+    let publishUrl = mainPath.includes(':3000') ? mainPath.replace(':3000/', ':8001/custom') : mainPath;
+    publishUrl = mainPath.includes(':9002') ? mainPath.replace(':9002/', ':8001/custom') : `${mainPath}/custom`;
+    console.log(`the final url is ${publishUrl}`);
+    const makeUrl = `${publishUrl}/update_samples`;
+    const delUrl = `${publishUrl}/delete_samples`;
     const queryTimeStamps = gql`
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/samples/EditSampleForm.tsx:52]: reuse the code pls

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/samples/EditSampleForm.tsx [hunk 7]
```diff
@@ -5,7 +5,7 @@ import { gql, useLazyQuery, useQuery } from '@apollo/client';
 import { useBaseEntity } from '../../../EntityContext';
 import { GetDatasetQuery } from '../../../../../../graphql/dataset.generated';
 import { FindMyUrn, FindWhoAmI, GetMyToken } from '../../../../dataset/whoAmI';
-import adhocConfig from '../../../../../../conf/Adhoc';
+// import adhocConfig from '../../../../../../conf/Adhoc';
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/samples/EditSampleForm.tsx:8]: remove unused imports

**Review by neojunjie — COMMENTED** [review]
It is good to move the wacky into a method/function. Easier for us to change the code in the future.

**Review by neojunjie — APPROVED** [review]

**Pull request merged**